src/modules/UI/scenes/SendConfirmation/action.js

```diff
@@ -1,82 +1,88 @@
 // @flow
-import {Actions} from 'react-native-router-flux'
-import {openABAlert} from '../../components/ABAlert/action'
-import * as Constants from '../../../../constants/indexConstants'
-import * as CORE_SELECTORS from '../../../Core/selectors.js'
-import * as UI_SELECTORS from '../../../UI/selectors.js'
-import * as WALLET_API from '../../../Core/Wallets/api.js'
+import { Actions } from 'react-native-router-flux'
+import { openABAlert } from '../../components/ABAlert/action'
+import { OPEN_AB_ALERT } from '../../../../constants/indexConstants'
+import { getWallet } from '../../../Core/selectors.js'
+import { getSelectedWalletId } from '../../selectors.js'
+import { type AbcMakeSpendInfo, getTransaction, getSpendInfo } from './selectors'
+import {
+  getMaxSpendable,
+  signTransaction,
+  broadcastTransaction,
+  saveTransaction,
+  makeSpend
+} from '../../../Core/Wallets/api.js'
 import type {
   AbcParsedUri,
-  AbcSpendInfo,
   AbcTransaction,
-  AbcCurrencyWallet,
-  AbcSpendTarget
+  AbcCurrencyWallet
 } from 'airbitz-core-types'
 
 const PREFIX = 'UI/SendConfimation/'
 
-export const UPDATE_AMOUNT_SATOSHI = PREFIX + 'UPDATE_AMOUNT_SATOSHI'
 export const UPDATE_LABEL = PREFIX + 'UPDATE_LABEL'
-export const UPDATE_MAX_AVAILABLE_TO_SPEND_IN_CRYPTO = PREFIX + 'UPDATE_MAX_AVAILABLE_TO_SPEND_IN_CRYPTO'
-export const ENABLE_SLIDER = PREFIX + 'ENABLE_SLIDER'
-export const UPDATE_DRAFT_STATUS = PREFIX + 'UPDATE_DRAFT_STATUS'
 export const UPDATE_IS_KEYBOARD_VISIBLE = PREFIX + 'UPDATE_IS_KEYBOARD_VISIBLE'
-export const UPDATE_FEE = PREFIX + 'UPDATE_FEE'
-export const UPDATE_MAX_SATOSHI = PREFIX + 'UPDATE_MAX_SATOSHI'
 export const UPDATE_SPEND_PENDING = PREFIX + 'UPDATE_SPEND_PENDING'
-
-export const UPDATE_WALLET_TRANSFER = PREFIX + 'UPDATE_WALLET_TRANSFER'
-export const UPDATE_PUBLIC_ADDRESS = PREFIX + 'UPDATE_PUBLIC_ADDRESS'
-export const UPDATE_SPEND_INFO = PREFIX + 'UPDATE_SPEND_INFO'
 export const RESET = PREFIX + 'RESET'
-
-export const UPDATE_CRYPTO_AMOUNT_REQUEST = PREFIX + 'UPDATE_CRYPTO_AMOUNT_REQUEST'
-export const USE_MAX_CRYPTO_AMOUNT = PREFIX + 'USE_MAX_CRYPTO_AMOUNT'
 export const UPDATE_PARSED_URI = PREFIX + 'UPDATE_PARSED_URI'
 export const UPDATE_TRANSACTION = PREFIX + 'UPDATE_TRANSACTION'
 
-export const UPDATE_NATIVE_AMOUNT = PREFIX + 'UPDATE_NATIVE_AMOUNT'
-export const CHANGE_MINING_FEE = PREFIX + 'CHANGE_MINING_FEE'
+export const updateWalletTransfer = (wallet: AbcCurrencyWallet) =>
+  (dispatch: any) => dispatch(updateLabel(wallet.name))
```

Comment:
This is a waste of a dispatch. Why is this even here when all it does is call updateLabel.

Reply:
Same is true for `updateParsedUri` and `updateMiningFees` below